### PR TITLE
Added auto focus to Github URL on opening of debug logs modal

### PR DESCRIPTION
**Pull request by ShivamSpm**
## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Fixes [vector-im/element-web/10822](https://github.com/vector-im/element-web/issues/10822)

Used react refs to add the auto focus to Github URL on opening the Submit debug logs modal

Before:

Github_URL-Issue-10822-Before

After (Result):

Github_URL-Issue-10822

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Added auto focus to Github URL on opening of debug logs modal ([\#10479](https://github.com/matrix-org/matrix-react-sdk/pull/10479)). Contributed by @ShivamSpm.

### CODE REVIEW

**Comment by t3chguy** [comment]
Can we not simplify this by just using the `autoFocus` attribute?

**Comment by t3chguy** [comment]
> Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md?rgh-link-date=2023-03-29T22%3A18%3A09Z))

Where is your sign off?

**Comment by ShivamSpm** [comment]
> > Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md?rgh-link-date=2023-03-29T22%3A18%3A09Z))
> 
> Where is your sign off?

I have added the sign off in this PR comments. I forgot to add it to my commit, sorry about that. I will add it in the commits from now on

**Comment by ShivamSpm** [comment]
> Can we not simplify this by just using the `autoFocus` attribute?

The autoFocus doesn't work here as this attribute is inconsistent. I searched a little bit and many people are facing issues with this attribute as it only works on the first render (Issue link -> [react/issues/5534](https://github.com/facebook/react/issues/5534)). Hence, this attribute works inconsistently in different components. Its better to use Refs to manage the state of the current component

File: src/components/views/dialogs/BugReportDialog.tsx [hunk 1]
```diff
@@ -53,9 +53,10 @@ interface IState {
 
 export default class BugReportDialog extends React.Component<IProps, IState> {
     private unmounted: boolean;
-
+    private issueRef: any;
```
Comment (t3chguy) [inline src/components/views/dialogs/BugReportDialog.tsx:56]: This needs a type specified

**Review by t3chguy — CHANGES_REQUESTED** [review]

**Review by t3chguy — APPROVED** [review]
Looks good to me, thanks for your contribution!

**Pull request merged**